### ISSUE TITLE
Chrome & Selects with OptGroup

### ISSUE THREAD

**Issue by ben-demotic** [tailwindlabs/tailwindcss#3104]
### Describe the problem:

Create a basic form with a select element on a page, which has a lot of options in an option group (example below has 1000).

Open the page in Firefox, it loads instantaneously.

Opened in Chrome or new Edge, the page takes several minutes to load.

The playground example also takes a while to load. 

The Github link below is to a full HTML test file which pulls Tailwind from a CDN.

The appears to be an issue with Tailwind, Chrome and optgroup.  If you remove Tailwind, Chrome renders the file instantaneously.  Alternatively, if you remove the optgroup tags, Chrome renders the file instantaneously.

The issue seems to be present in 1.9.6 and the latest version of 2.

Chrome is 87.0.4280.88 on Windows 10 (up to date).
Firefox is 84.0 on Windows 10.

### Link to a minimal reproduction:

https://play.tailwindcss.com/QX9Q7RuAqo

https://github.com/ben-demotic/examples/blob/main/tailwind_optgroup_issue.html

**Comment by adamwathan** [comment]
Wow that is a weird one! I just hopped on a call with @argyleink on the Chrome team to work through it and try to pinpoint the problem and really no idea just seems like a Chrome bug. He's going to write up a bug report for the Chrome team and hopefully someone who knows the internals better will have a lightbulb moment about what could be causing this.

**Comment by argyleink** [comment]
Here's the bug reported https://bugs.chromium.org/p/chromium/issues/detail?id=1159206, please comment additional information! 👍🏻 

Will touch base back here when I've found more from the engineering team ⌚

**Comment by argyleink** [comment]
confirmed, Chrome bug:

**tldr;**
- `.space-y-0 > :not([hidden]) ~ :not([hidden])` is expensive
- this problem = expensive selector + loops that dont escape early
- aka not y'all's problem
- we found 2 bugs related to it and optgroup, engineers are set to hack on it
- we should be able to fix it without changes to y'alls code
- i'll ping back when chrome wont hang the page on load 👍🏻

**Comment by argyleink** [comment]
https://chromium-review.googlesource.com/c/chromium/src/+/2595379/

^ fixes the issue. pending tests and all the builds pass, fix is imminent though 👍🏻

**Comment by adamwathan** [comment]
Amazing, thanks @argyleink 🙌🏻

Going to close this since the fix is out of our control and has been addressed by the Chromium team.

**Comment by ben-demotic** [comment]
Many thanks to @adamwathan and @argyleink for their extremely quick response on this one, much appreciated 👍
